Week 1/AtoI:ItoA/Solution.java

```diff
@@ -0,0 +1,59 @@
+import java.io.*;
+import java.util.*;
+
+public class Solution {
+	public static int atoi(String s){
+		int sign = 1;
+		int n = s.length();
+		if(n==0) throw new NumberFormatException();
+		if(s.charAt(0)==45) sign = -1;
+		int i = 0; int res = 0;
+		if(sign==-1) i = 1;
+		for(;i<n;i++){
+			int cur = s.charAt(i) - 48;
+			if(cur>9||cur<0) throw new NumberFormatException();
+			res = res*10+cur;
+			if(res%10!=cur) throw new NumberFormatException();
```

Comment:
Yes, I use that line to throw an error in case the number (string) is larger than what the int type allows